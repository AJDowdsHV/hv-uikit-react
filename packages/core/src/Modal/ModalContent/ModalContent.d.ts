import { DialogContentProps, StandardProps } from "@material-ui/core";

export interface HvModalContentProps
  extends StandardProps<DialogContentProps, HvModalContentClassKey> {}

export type HvModalContentClassKey = "root" | "textContent";

export default function HvModalContent(props: HvModalContentProps): JSX.Element | null;
