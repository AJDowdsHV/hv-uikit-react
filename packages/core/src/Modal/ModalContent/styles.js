const styles = theme => ({
  root: {
    padding: `${theme.hv.spacing.xs}px ${theme.hv.spacing.sm}px ${theme.hv.spacing.sm}px ${theme.hv.spacing.sm}px`,
    flex: "none"
  },
  textContent: {
    marginLeft: "50px"
  }
});

export default styles;
