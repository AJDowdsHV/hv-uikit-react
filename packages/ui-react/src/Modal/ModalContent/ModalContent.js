/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import React from "react";
import PropTypes from "prop-types";
import MuiDialogContent from "@material-ui/core/DialogContent";
import classNames from "classnames";
import Typography from "@material-ui/core/Typography";

/**
 * Message container. The passed children is render in this container, creating
 * a typography node if the children is of the string type.
 *
 * @param classes
 * @param children
 * @returns {*}
 * @constructor
 */
const ModalContent = ({ classes, children }) => {
  const isString = typeof children === "string";

  return (
    <MuiDialogContent
      className={classNames(classes.root, {
        [classes.textContent]: isString
      })}
    >
      {!isString && children}
      {isString && <Typography variant="body1">{children}</Typography>}
    </MuiDialogContent>
  );
};
ModalContent.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Style applied to the root of the component (container for the content).
     */
    root: PropTypes.instanceOf(Object),
    /**
     * Style applied when the content is a string.
     */
    textContent: PropTypes.instanceOf(Object)
  }).isRequired,
  /**
   * Content to be render.
   */
  children: PropTypes.node.isRequired
};

export default ModalContent;
