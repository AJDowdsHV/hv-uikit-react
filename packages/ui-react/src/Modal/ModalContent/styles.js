/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

const styles = theme => ({
  root: {
    padding: ` ${theme.spacing.md}px 0px 0px 0px`,
    flex: "none"
  },
  textContent: {
    marginLeft: "58px"
  }
});

export default styles;
