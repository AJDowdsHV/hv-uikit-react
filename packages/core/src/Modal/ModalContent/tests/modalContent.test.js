/* eslint-env jest */

import React from "react";
import { shallow } from "enzyme";

import HvProvider from "../../../Provider";
import ModalContent from "..";

describe("[v3] ModalContent withStyles", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider>
        <ModalContent>Modal Content</ModalContent>
      </HvProvider>
    );
  });

  it("should render correctly", () => {
    expect(wrapper.find(ModalContent)).toMatchSnapshot();
  });
});

describe("[v3] ModalContent Component", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider>
        <ModalContent>Modal Content</ModalContent>
      </HvProvider>
    );
  });

  it("should render correctly if opened", () => {
    expect(wrapper.find(ModalContent)).toMatchSnapshot();
  });
});
