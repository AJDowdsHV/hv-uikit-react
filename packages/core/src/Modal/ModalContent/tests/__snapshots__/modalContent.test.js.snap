// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent Component should render correctly if opened 1`] = `
<WithStyles(ModalContent)>
  Modal Content
</WithStyles(ModalContent)>
`;

exports[`ModalContent withStyles should render correctly 1`] = `
<WithStyles(ModalContent)>
  Modal Content
</WithStyles(ModalContent)>
`;
